src/noisepy/seis/datatypes.py

```diff
@@ -157,3 +157,16 @@ def __init__(self, stream: obspy.Stream):
         self.data = stream[0].data
         self.sampling_rate = stream[0].stats.sampling_rate
         self.start_timestamp = stream[0].stats.starttime.timestamp
+
+
+@dataclass
+class NoiseFFT:
```

Comment:
@mdenolle - some naming questions:
- Is there a better name for this class? 
- Is `N` the number of segments? If so I'd like to rename it to something line `SegmentCount`. 
- Looks like `Nfft2` is half the spectrum size? If so I'd prefer to return the full size, call it  `SpectrSz` and have the caller half-it as needed.

Reply:
N is indeed the number of segments or the count of windows. I see in S1 that we also use "nseg_chunk".

The time series length is ``fft_params.cc_len * fft_params.samp_freq``. We fill with zeros before doing the FFT up to ``nnfft`` because it speeds up the FFT.

``NFFT2`` should be ``nnfft//2``. In a way, i agree that we don't need to call. But we should not store the second half of the FFT and only keep in memory the data up to nnfft//2.